service/featuregate/flags.go

```diff
@@ -0,0 +1,96 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package featuregate // import "go.opentelemetry.io/collector/service/featuregate"
+
+import (
+	"flag"
+	"sort"
+	"strings"
+)
+
+const gatesListCfg = "feature-gates"
+
+var gatesList = FlagValue{}
+
+// Flags adds CLI flags for managing feature gates to the provided FlagSet
+// Feature gates can be configured with `--feature-gates=foo,-bar`.  This would
+// enable the `foo` feature gate and disable the `bar` feature gate.
+func Flags(flags *flag.FlagSet) {
+	flags.Var(
+		gatesList,
+		gatesListCfg,
+		"Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature.  '+' or no prefix will enable the feature.")
+}
+
+// GetFlags returns the FlagValue used with Flags()
+func GetFlags() FlagValue {
+	return gatesList
+}
+
+var _ flag.Value = (*FlagValue)(nil)
+
+// FlagValue implements the flag.Value interface and provides a mechanism for applying feature
+// gate statuses to a Registry
+type FlagValue map[string]bool
+
+// String returns a string representing the FlagValue
+func (f FlagValue) String() string {
```

Comment:
Does this need to be public? (I want to understand)

Reply:
Yes.  It is part of the [`flag.Value` interface](https://pkg.go.dev/flag#Value).

Reply:
Sorry wrong line of comment, was looking at SetSlice (you can answer on Slack or after the merge).